src/core/src/node.cpp

```diff
@@ -75,24 +75,27 @@ ov::Node::Node(const OutputVector& arguments, size_t output_size) : Node() {
 }
 
 ov::Node::~Node() {
-    // raise a flag to reset nodes cache
-    for_each(m_shared_rt_info.cbegin(), m_shared_rt_info.cend(), [](const std::shared_ptr<SharedRTInfo>& info) {
-        info->set_use_topological_cache(false);
-    });
-
-    for (descriptor::Input& input : m_inputs) {
-        if (input.has_output()) {
-            // This test adds 1 to the actual count, so a count of 2 means this input is the only
-            // reference to the node.
-            if (input.get_output().get_node().use_count() == 2) {
-                // Don't want to trigger a deep recursive delete
-                NodeVector nodes{input.get_output().get_node()};
+    try {
+        // raise a flag to reset nodes cache
+        for_each(m_shared_rt_info.cbegin(), m_shared_rt_info.cend(), [](const std::shared_ptr<SharedRTInfo>& info) {
+            info->set_use_topological_cache(false);
+        });
+
+        for (descriptor::Input& input : m_inputs) {
+            if (input.has_output()) {
+                // This test adds 1 to the actual count, so a count of 2 means this input is the only
+                // reference to the node.
+                if (input.get_output().get_node().use_count() == 2) {
+                    // Don't want to trigger a deep recursive delete
+                    NodeVector nodes{input.get_output().get_node()};
+                    input.remove_output();
+                    safe_delete(nodes, true);
+                    return;
+                }
                 input.remove_output();
-                safe_delete(nodes, true);
-                return;
             }
-            input.remove_output();
         }
+    } catch (...) {
```

Comment:
Maybe add 'std::err' and exit here? It looks abnormal situation if exception occurs here, there looks no way to properly recover

Reply:
Library shouldn't print any to user output

Reply:
I'm not sure such broad exception handling is the right way because any next code extension will be also handled inside try/except. Perhaps it would be better to update `ov::Node::safe_delete()` so that the function may be marked as `noexcept`

Reply:
Here we have 2 places which thow exceptions. `save_delete()` (which calls `get_output().get_node()`) and `get_output().get_node()` inside the destructor.
If `save_delete()` we can mark as `noecept`, but I don't want to mark as `noexcept` `get_output().get_node()` method.

I don't see any issue to have one handler for exception from these methods.